setup.cfg

```diff
@@ -1,5 +1,21 @@
 [flake8]
-exclude = build,.git,build-pypy,.tox
+exclude =
+    .git,
+    __pycache__,
+    docs/source/conf.py,
+    docs/conf.py,
```

Comment:
docs/conf.py is flake8 compliant, why add it to the exclude list?

Reply:
Because he will still check it, but the content of `conf.py` does not meet the EPE8 specification, so Flake8, Pylint, MyPy, etc. still check this directory.

In addition, I suggest that all non-Python directories and folders be added here, so that the CI part can be targeted at the specified directory, reducing the ignorance of some Python programs.

Reply:
conf.py in Dulwich does meet the PEP8 specification:

% flake8 docs/conf.py 
%

so why exclude it?

Reply:
Yes, sorry, I remembered it wrong, it was `Pylint`.

```
timmy@timmy-Latitude-E5550:~/Git/dulwich$ python3 -m flake8 docs/conf.py 
timmy@timmy-Latitude-E5550:~/Git/dulwich$ python3 -m pylint docs/conf.py 
************* Module conf
docs/conf.py:52:0: W0622: Redefining built-in 'copyright' (redefined-builtin)
docs/conf.py:1:0: C0114: Missing module docstring (missing-module-docstring)
docs/conf.py:36:0: C0103: Constant name "autoclass_content" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:42:0: C0103: Constant name "source_suffix" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:48:0: C0103: Constant name "master_doc" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:51:0: C0103: Constant name "project" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:52:0: C0103: Constant name "copyright" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:59:0: C0103: Constant name "version" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:61:0: C0103: Constant name "release" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:96:0: C0103: Constant name "pygments_style" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:107:0: C0103: Constant name "html_theme" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:174:0: C0103: Constant name "htmlhelp_basename" doesn't conform to UPPER_CASE naming style (invalid-name)

-----------------------------------
Your code has been rated at 4.29/10

timmy@timmy-Latitude-E5550:~/Git/dulwich$ python3 -m flake8 docs/conf.py 
timmy@timmy-Latitude-E5550:~/Git/dulwich$ python3 -m pylint docs/conf.py 
************* Module conf
docs/conf.py:52:0: W0622: Redefining built-in 'copyright' (redefined-builtin)
docs/conf.py:1:0: C0114: Missing module docstring (missing-module-docstring)
docs/conf.py:36:0: C0103: Constant name "autoclass_content" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:42:0: C0103: Constant name "source_suffix" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:48:0: C0103: Constant name "master_doc" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:51:0: C0103: Constant name "project" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:52:0: C0103: Constant name "copyright" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:59:0: C0103: Constant name "version" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:61:0: C0103: Constant name "release" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:96:0: C0103: Constant name "pygments_style" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:107:0: C0103: Constant name "html_theme" doesn't conform to UPPER_CASE naming style (invalid-name)
docs/conf.py:174:0: C0103: Constant name "htmlhelp_basename" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 4.29/10 (previous run: 4.29/10, +0.00)

timmy@timmy-Latitude-E5550:~/Git/dulwich$ 
```

Reply:
In addition, I refer to here, so if I have an error or describe the problem, you can give me feedback.

https://flake8.pycqa.org/en/latest/user/configuration.html#project-configuration